### PR TITLE
Allow referencing type-only exports as namespace members in ImportTypes and TypeQueries

### LINKED ISSUE #45946: Type-only export causes `no exported member` error in generated `.d.ts` file
# Bug Report

### 🔎 Search Terms

"export type"
typeof
"has no exported member"

### 🕗 Version & Regression Information

This is the behavior I observed in versions: 4.5.0-dev.20210917, 4.4.3, 4.3.5, 4.2.4, 4.1.6
It did not occur in versions: 4.0.8, 3.9.10, 3.8.3

### ⏯ Playground Link

Sorry, I could not find a way to reproduce this in the bug workbench. Here is an [attempt](https://www.typescriptlang.org/dev/bug-workbench/?ts=4.5.0-dev.20210910#code/PTAEAEDMEsBsFMB2BDAtvAXKARgVwC74D2iAdPgM4BQ0qADkQE76gDeAxrMhRQLLJ0AvqEiMiqUAHJSwWNHykAVhUkBuKvAAeDZqHYkKLdqAC8erj350AFAEp1VEBBgIU6LHIWUqnbhVAAwhZ8AgAi0Izw7PjQAG7wbIJUGtpMLPgAnnQJrEF+VuGR0XHwgg5aOkYGLAAmEVEx8aZUAAWgADwBAHzWvjwA+lgBtqZdraB2oxOs4211RY2Y5n79ADTjgvZAA), but I could not find a way to disable angle brackets from being interpreted as JSX. Additionally, the error occurs when compiling generated `.d.ts` output, rather than in the initial compile pass.

### 💻 Code

```ts
// @filename: button.ts
import {classMap} from './lit.js';
export const c = classMap();
```

```ts
// @filename: lit.ts
class ClassMapDirective {}

export type {ClassMapDirective};

export const directive =
  <C>(class_: C) =>
  () => ({
    directive: class_,
  });
```

The problem occurs in the generation of the `.d.ts` file for `button.ts`. The output is:

```ts
// @filename: button.d.ts
export declare const c: {
    directive: typeof import("./lit.js").ClassMapDirective;
};
```

### 🙁 Actual behavior

Compiling the generated `button.d.ts` file results in an error:

```
button.d.ts:2:42 - error TS2694: Namespace '"/home/aomarks/scratch/tsbug/lit"' has no exported member 'ClassMapDirective'.

2     directive: typeof import("./lit.js").ClassMapDirective;
                                           ~~~~~~~~~~~~~~~~~
```

### 🙂 Expected behavior

The generated `button.d.ts` file should compile without errors.

One thing I noticed is that removing the `typeof` operator from the generated `button.d.ts` file seems to fix the error. I wonder if this bug is related to that. Could it be that a symbol which is exported with a type-only export is *already* a type, unlike a normal export, so the use of a `typeof` operator is redundant and invalid, since `typeof` expects the LHS to be a value?

### PR DESCRIPTION
**Pull request by andrewbranch**
Fixes #45946

### CODE REVIEW

File: tests/baselines/reference/typeofImportTypeOnlyExport.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+export type { ClassMapDirective };
+export declare const directive: <C>(class_: C) => () => {
+    directive: C;
+};
+export declare const classMap: () => {
+    directive: typeof ClassMapDirective;
+};
+//// [button.d.ts]
+export declare const c: {
+    directive: typeof import("./lit.js").ClassMapDirective;
```
Comment (andrewbranch) [inline tests/baselines/reference/typeofImportTypeOnlyExport.js:57]: This is what declaration emit has always done, but was an error until now.

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -28997,7 +28999,7 @@ namespace ts {
                     }
                     return isErrorType(apparentType) ? errorType : apparentType;;
                 }
-                prop = getPropertyOfType(apparentType, right.escapedText);
+                prop = getPropertyOfType(apparentType, right.escapedText, /*skipObjectFunctionPropertyAugment*/ false, /*includeTypeOnlyMembers*/ node.kind === SyntaxKind.QualifiedName);
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts:29002]: Nit: is there a reason why here you used `/*skipObjectFunctionPropertyAugment*/ false` but above you use `undefined`?

File: src/compiler/checker.ts [hunk 3]
```diff
@@ -28997,7 +28999,7 @@ namespace ts {
                     }
                     return isErrorType(apparentType) ? errorType : apparentType;;
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts:29000]: `;;` :(

**Review by DanielRosenwasser — APPROVED** [review]
Seems reasonable. I'm surprised it was that that easy to plumb through.

File: src/compiler/checker.ts (hunk 3 above)
Comment (DanielRosenwasser) [reply src/compiler/checker.ts:29000]: I sent a quick fix for this.

**Review by weswigham — APPROVED** [review]

File: src/compiler/checker.ts (hunk 2 above)
Comment (andrewbranch) [reply src/compiler/checker.ts:29002]: Just to see if you were paying attention

File: src/compiler/checker.ts (hunk 3 above)
Comment (andrewbranch) [reply src/compiler/checker.ts:29000]: I don’t recall seeing this or touching this line. Weird.

Comment (andrewbranch) [reply src/compiler/checker.ts:29000]: Oh, it’s because you did

**Pull request merged**